tapgarden/planter.go

```diff
@@ -468,38 +470,19 @@ func (c *ChainPlanter) gardener() {
 	for {
 		select {
 		case <-c.cfg.BatchTicker.Ticks():
-			// No pending batch, so we can just continue back to
-			// the top of the loop.
+			// There is no pending batch, so we can just abort.
```

Comment:
Also, this interface for finalizing a batch is now async while the `FinalizeBatch` call is now sync?

Reply:
Finalizing a batch was never sync in the first place. We just fired off the caretaker and never waited for any response. So errors would only ever be logged in the log and not returned to the caller.
But yes, the `c.cfg.BatchTicker.Ticks()` is still async, as there is no "caller" to return a status to. And since this path currently only is taken on daemon startup, this isn't really used anymore (maybe we should also remove that path and completely remove the ticker after all? since IMO batch creation should be explicit).

Reply:
AFAICT the ticker is only directly accessed in the unit tests for `tickMintingBatch`, and indeed does not need to be exposed any more. IIUC there is also no consequence from moving the `finalizeBatch` call into the inner `switch{}` statement? If so i'm ok with removing the ticker and that case to handle `Ticks()` entirely.

Didn't think about restart, but I agree that properly saving batch state is the priority for crash safety. Can turn off resume-on-restart here:
https://github.com/lightninglabs/taproot-assets/blob/c5bdbfe25a6c5485ee15be0d91da3af80e3b7493/tapgarden/planter.go#L288

Reply:
Tick was added initially to support optimistic aggregation. Now we have the explicit batch creation, so can safely be removed if we don't see any value at all in it.